src/lib/storage/fitted_attribute_vector.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+
+#include "base_attribute_vector.hpp"
+
+namespace opossum {
+
+template <typename uintX_t>
+class FittedAttributeVector : public BaseAttributeVector {
+ public:
+  explicit FittedAttributeVector(const size_t size) : BaseAttributeVector(), _values(size) {}
+
+  ValueID get(const size_t i) const override { return static_cast<ValueID>(_values.at(i)); }
+
+  // inserts the value_id at a given position
+  void set(const size_t i, const ValueID value_id) override { _values.at(i) = static_cast<uintX_t>(value_id); }
```

Comment:
Is the static_cast needed here? Same goes for the get method

Reply:
In the method above it can be removed. But here I believe it makes sense, as it indicates that we are not necessarily storing the `ValueID` at a full 32bit length.